test/abstractarray.jl

```diff
@@ -910,6 +910,9 @@ include("generic_map_tests.jl")
 generic_map_tests(map, map!)
 @test_throws ArgumentError map!(-, [1])
 
+# Issue #30624
+@test map!(+, [0,0,0], [1,2], [10,20,30], [100]) == [111,0,0]
```

Comment:
Would you also add a test for https://github.com/JuliaLang/julia/issues/36235#issuecomment-2503895662, please?